mllib/src/main/scala/org/apache/spark/ml/util/ReadWrite.scala

```diff
@@ -300,15 +301,23 @@ private[ml] object DefaultParamsWriter {
       paramMap: Option[JValue] = None): String = {
     val uid = instance.uid
     val cls = instance.getClass.getName
-    val params = instance.extractParamMap().toSeq.asInstanceOf[Seq[ParamPair[Any]]]
+    val params = instance.extractParamMap().toSeq
+      .filter(_.param.name != "initialModel").asInstanceOf[Seq[ParamPair[Any]]]
```

Comment:
Is it not possible to check if the param is an instance of `Param[Model[_]]` rather than the name?